src/main/java/seedu/address/logic/parser/FilterCommandParser.java

```diff
@@ -0,0 +1,97 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_GOAL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INGREDIENT_GRAIN;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INGREDIENT_OTHER;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INGREDIENT_PROTEIN;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INGREDIENT_VEGE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TIME;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.logic.commands.FilterCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.goal.Goal;
+import seedu.address.model.recipe.RecipeMatchesKeywordsPredicate;
+import seedu.address.model.recipe.Time;
+
+/**
+ * Parses input arguments and creates a new FilterCommand object
+ */
+public class FilterCommandParser implements Parser<FilterCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FilterCommand
+     * and returns a FilterCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FilterCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_TIME, PREFIX_INGREDIENT_GRAIN, PREFIX_INGREDIENT_VEGE,
+                        PREFIX_INGREDIENT_PROTEIN, PREFIX_INGREDIENT_OTHER, PREFIX_GOAL);
+
+        List<Time> filterByTime;
+        if (argMultimap.getValue(PREFIX_TIME).isPresent()) {
+            filterByTime = parseTimeForFilter(argMultimap.getValue(PREFIX_TIME).get());
+        } else {
+            filterByTime = Collections.emptyList();
+        }
+
+        Set<Goal> filterByGoals;
+        if (!argMultimap.getAllValues(PREFIX_GOAL).isEmpty()) {
+            filterByGoals = parseGoalsForFilter(argMultimap.getAllValues(PREFIX_GOAL));
+            System.out.println(argMultimap.getAllValues(PREFIX_GOAL));
```

Comment:
Might want to remove this!

Reply:
LOL thanks for pointing it out! Removed :)